website/client/src/components/tasks/spells.vue

```diff
@@ -267,18 +270,14 @@ export default {
       );
     },
     spellDisabled (skill) {
-      if (skill === 'frost' && this.user.stats.buffs.streaks) {
-        return true;
-      }
-      // @TODO: Implement
-      // } else if (skill === 'stealth' && this.user.stats.buffs.stealth
-      // >= this.user.dailys.length) {
-      //   return true;
-      // }
+      const incompleteDailiesDue = this.getUnfilteredTaskList('daily').filter(daily => !daily.completed && daily.isDue).length;
+
+      if (skill === 'frost' && this.user.stats.buffs.streaks) return true;
+      if (skill === 'stealth' && this.user.stats.buffs.stealth >= incompleteDailiesDue) return true;
 
       return false;
     },
-    // @TODO is this used?
+
```

Comment:
So I just tested this PR locally and while the code works fine I think that `spellRogueStealthDaliesAvoided` will always show that 0 dailies will be avoided because `this.user.stats.buffs.stealth ` ( see lines 289-292 of this file ) is increased after the skill is casted not before. Am I correct @SabreCat @Alys ?

Reply:
Hmm my followup questions would be - in the code for casting Stealth, the dailies are calculated using `user.tasksOrder.dailys.length` and we discussed above that in some cases this may not include the right number of tasks. What kinds of cases are those?
https://github.com/HabitRPG/habitica/blob/b5182ed7d454f0a01497da8472b54d642fbe43ee/website/common/script/content/spells.js#L212

If stealth increases only after the skill is cast, it seems like we'd want to make the following changes
- have a common method in spell.js that calculates the stealth buff to be added (lines 211 and 212 of spells.js)
- refactor spells.js to use the method --> user.stats.buffs.stealth +=  stealthBuffToBeAdded()
- for the stealth note previewing num dailies to be avoided, use the return value of stealthBuffToBeAdded() instead of the current stealth buff

Looking at git history though, using `this.user.stats.buffs.stealth` for that tooltip has been in place for the last 3 years. So is this a long standing bug or is there some context we're missing? 😅

Reply:
@paglias I think you're correct.

@alexandreabeh I think it's long standing bug, alas. :-/ 
From what I can tell, the answer to that comment `// @TODO is this used?` has been No for possibly the last 3 years. :laughing:   I've checked the `frost` skill and it's special note isn't used either.

From my point of view, the approach you've described looks good but wait for paglias to comment and listen to him not me if he has other suggestions. :)

Reply:
that approach looks good @alexandreabeh !

Reply:
@paglias I've pushed changes with the approach we just talked about. I was able to see the '0' dailies behavior you were talking about, and now the tooltip gives the same preview number as whatever is calculated during the actual casting.

I decided to leave the dailies length code alone in the spell.js file after looking at the [Rogue wiki](https://habitica.fandom.com/wiki/Rogue#Stealth) and realizing the calculation there is supposed to include both complete and incomplete dailies. So that's different from the way we need to look at incomplete + due dailies for whether the spell should be disabled.